Managers/ActivityMan.cpp

```diff
@@ -1,1852 +1,164 @@
+		g_InActivity = true;
 
-void Activity::ChangeTeamFunds(float howMuch, unsigned int which)
-{
-	if (which >= Activity::TEAM_1 && which < Activity::MAXTEAMCOUNT)
-	{
-		m_TeamFunds[which] += howMuch;
-		m_FundsChanged[which] = true;
-		if (IsPlayerTeam(which))
-			g_GUISound.FundsChangedSound()->Play();
-	}
-}
+		g_PostProcessMan.ClearScenePostEffects();
+		g_FrameMan.ClearScreenText();
 
+		// Reset the mouse input to the center
+		g_UInputMan.SetMouseValueMagnitude(0.05F);
 
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          TeamFundsChanged
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Checks whetehr the team funds changed since last time this was called.
-//                  This also resets the state, so calling this again on the same team will
-//                  ield false unless it's been changed again.
+		// Reset the last music position
+		m_LastMusicPath = "";
+		m_LastMusicPos = 0;
 
-bool Activity::TeamFundsChanged(unsigned int which)
-{
-	if (which >= Activity::TEAM_1 && which < Activity::MAXTEAMCOUNT)
-	{
-		bool changed = m_FundsChanged[which];
-		m_FundsChanged[which] = false;
-		return changed;
+		return error;
 	}
-	return false;
-}
-
 
-//////////////////////////////////////////////////////////////////////////////////////////
-// Virtual method:  EnteredOrbit
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Indicates an Actor as having left the game scene and entered orbit.
-//                  OWNERSHIP IS NOT transferred, as the Actor's inventory is just 'unloaded'.
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-void Activity::EnteredOrbit(Actor *orbitedCraft)
-{
-    if (!orbitedCraft)
-        return;
+	int ActivityMan::StartActivity(string className, string instanceName) {
+		const Entity *entity = g_PresetMan.GetEntityPreset(className, instanceName);
 
-    int team  = orbitedCraft->GetTeam();
-/*
-    if (m_pOrbitRocket[team])
-    {
-        delete m_pOrbitRocket[team];
-        m_pOrbitRocket[team] = 0;
-    }
-*/
-
-	// Find out cost multipliers for metagame, or we end up AI's 
-	// earning money when their crafts return if their nativeCostMultiplier is too low
-    float foreignCostMult = 1.0;
-	float nativeCostMult = 1.0;
-	if (g_MetaMan.GameInProgress())
-	{
-		for (int player = 0 ; player < MAXPLAYERCOUNT; player++)
-		{
-			if (GetTeamOfPlayer(player) == orbitedCraft->GetTeam())
-			{
-				MetaPlayer *pMetaPlayer = g_MetaMan.GetMetaPlayerOfInGamePlayer(player);
-				if (pMetaPlayer)
-				{
-					foreignCostMult = pMetaPlayer->GetForeignCostMultiplier();
-					nativeCostMult = pMetaPlayer->GetNativeCostMultiplier();
-				}
+		if (entity) {
+			Activity *newActivity = dynamic_cast<Activity *>(entity->Clone());
+			if (newActivity) {
```

Comment:
Why would we need it here?
`StartActivity` will assert if the passed in activity is screwed up.

Reply:
dynamic_cast: `If the cast is successful, dynamic_cast returns a value of type new-type. If the cast fails and new-type is a pointer type, it returns a null pointer of that type.`

If we fail to cast the activity with dynamic_cast then newActivity will be a null pointer. This means the thing that tries to StartActivity will never happen and it'll silently fail instead of trying to call `StartActivity(Activity *)`. I'm also fine with removing the if check and letting the assert trigger, I just figured a lua error would be less harsh if it happens. But then again, it might be better to hard crash if this happens, cause it means presetman is all kinds of screwed up.